sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -642,7 +642,7 @@ case class RegExpReplace(subject: Expression, regexp: Expression, rep: Expressio
     }
     val source = s.toString()
     val position = i.asInstanceOf[Int] - 1
-    if (position < source.length) {
+    if (position == 0 || position < source.length) {
```

Comment:
How about `position <= source.length` ?

Reply:
That changes the logic; then this check would always be true?

Reply:
+1 to what @srowen says, the logic would be different then cause the evaluation of the position would change and trigger from one position earlier than it used to

Reply:
> That changes the logic; then this check would always be true?

For example, the source is "abc", the position need in range [0, 3].
If the source is "",  the position must be 0.

Reply:
Yeah, I got it. The logic looks good.